pkg/addonmanager/manager.go

```diff
@@ -321,10 +345,40 @@ func (a *addonManager) StartWithInformers(ctx context.Context,
 	return nil
 }
 
-// New returns a new Manager for creating addon agents.
-func New(config *rest.Config) (AddonManager, error) {
+// ManagerOptions defines the flags for addon manager
+type ManagerOptions struct {
+	WorkDriver          string
+	WorkDriverConfig    string
+	CloudEventsClientID string
+	SourceID            string
+}
+
+// NewManagerOptions returns the flags with default value set
+func NewManagerOptions() *ManagerOptions {
+	return &ManagerOptions{
+		// set default work driver to kube
+		WorkDriver: cloudeventswork.ConfigTypeKube,
+	}
+}
+
+// AddFlags adds the addon manager flags to the given command
+func (o *ManagerOptions) AddFlags(cmd *cobra.Command) {
+	flags := cmd.Flags()
+	flags.StringVar(&o.WorkDriver, "work-driver",
+		o.WorkDriver, "The type of work driver, currently it can be kube, mqtt or grpc")
+	flags.StringVar(&o.WorkDriverConfig, "work-driver-config",
+		o.WorkDriverConfig, "The config file path of current work driver")
+	flags.StringVar(&o.CloudEventsClientID, "cloudevents-client-id",
+		o.CloudEventsClientID, "The ID of the cloudevents client when publishing works with cloudevents")
+	flags.StringVar(&o.SourceID, "source-id",
+		o.SourceID, "The ID of the source when publishing works with cloudevents")
+}
+
+// New returns a new addon manager for creating addon agents.
+func New(config *rest.Config, opts *ManagerOptions) (AddonManager, error) {
```

Comment:
this will break the existing signature of the function. the opts filed should be an optional field. And we should allow that it is not set.

Reply:
updated this to make managerOptions be optional.